compiler/rustc_hir/src/def.rs

```diff
@@ -138,6 +140,7 @@ impl DefKind {
                 panic!("impossible struct constructor")
             }
             DefKind::OpaqueTy => "opaque type",
+            DefKind::ImplTraitPlaceholder => "opaque type in trait",
```

Comment:
not sure what a better description would be but it's not opaque when refined...not sure how confusing that will be.

Reply:
I don't think people usually see these def-kind names unless they're happening in errors, and I don't think you'd see this def-kind specifically being named unless it's _not_ being normalized, i.e. it's not refined. I don't think this needs changing unless someone has a particularly good replacement name.